bot/exts/moderation/stream.py

```diff
@@ -0,0 +1,135 @@
+import time
+
+import discord
+from async_rediscache import RedisCache
+from discord.ext import commands, tasks
+
+from bot.bot import Bot
+from bot.constants import Guild, Roles, STAFF_ROLES, TIME_FORMATS
+
+# Constant error messages
+NO_USER_SPECIFIED = "Please specify a user"
+TIME_FORMAT_NOT_VALID = "Please specify a valid time format ex. 10h or 1day"
+TIME_LESS_EQ_0 = "Duration can not be a 0 or lower"
+USER_ALREADY_ALLOWED_TO_STREAM = "This user can already stream"
+USER_ALREADY_NOT_ALLOWED_TO_STREAM = "This user already can't stream"
+
+
+# FORMATS holds a combined list of all allowed time units
+# made from TIME_FORMATS constant
+FORMATS = []
+for key, entry in TIME_FORMATS.items():
+    FORMATS.extend(entry["aliases"])
+    FORMATS.append(key)
+
+
+class Stream(commands.Cog):
+    """Stream class handles giving screen sharing permission with commands."""
+
+    # Data cache storing userid to unix_time relation
+    # user id is used to get member who's streaming permission need to be revoked after some time
+    # unix_time is a time when user's streaming permission needs tp be revoked in unix time notation
+    user_cache = RedisCache()
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.remove_permissions.start()
+        self.guild_static = None
+
+    @staticmethod
+    def _link_from_alias(time_format: str) -> (dict, str):
+        """Get TIME_FORMATS key and entry by time format or any of its aliases."""
+        for format_key, val in TIME_FORMATS.items():
+            if format_key == time_format or time_format in val["aliases"]:
+                return TIME_FORMATS[format_key], format_key
+
+    def _parse_time_to_seconds(self, duration: int, time_format: str) -> int:
+        """Get time in seconds from duration and time format."""
+        return duration * self._link_from_alias(time_format)[0]["mul"]
+
+    @commands.command(aliases=("streaming", "share"))
+    @commands.has_any_role(*STAFF_ROLES)
+    async def stream(
+            self,
+            ctx: commands.Context,
+            user: discord.Member = None,
+            duration: int = 1,
+            time_format: str = "h",
```

Comment:
Use the extant [`Duration`](https://github.com/python-discord/bot/blob/master/bot/converters.py#L301) converter instead of rolling your own. In fact, it'd be even better to use `Expiry` defined there too. This will support specifying exact timestamps too and make it consistent with the infraction commands.